packages/test/test-drivers/src/odspTestDriver.ts

```diff
@@ -107,22 +108,32 @@ function getCredentials(
 		for (let i = 0; i < range.count; i++) {
 			const username = `${range.prefix}${range.start + i}@${tenant}`;
 			if (requestedUserName === undefined || requestedUserName === username) {
-				creds[username] = range.password;
+				creds.push({ username, password: range.password });
 			}
 		}
 	} else {
 		const loginAccounts =
 			odspEndpointName === "odsp"
 				? process.env.login__odsp__test__accounts
 				: process.env.login__odspdf__test__accounts;
-		assert(loginAccounts !== undefined, "Missing login__odsp/odspdf__test__accounts");
+		if (loginAccounts === undefined) {
+			// eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+			const inCi = !!process.env.TF_BUILD;
+			const odspOrOdspdf = odspEndpointName === "odsp" ? "odsp" : "odspdf";
+			throw new Error(
```

Comment:
Sorry, why are we throwing error here?

Reply:
I could move back to assert.fail if you want. Not super opinionated.